pkg/controller/releasemanager/syncer_test.go

```diff
@@ -316,3 +316,30 @@ func TestPrepareRevisionsAndRulesIncompleteScaleUp(t *tt.T) {
 	revisions = testResourceSyncer.prepareRevisions()
 	assertIncarnationPercent(t, releasableIncarnations, revisions, []int{60, 40})
 }
+
+func TestFailingReleasedRevision(t *tt.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	m := createTestIncarnations(ctrl)
+	testResourceSyncer := &ResourceSyncer{
+		incarnations: m,
+		log:          test.MustNewLogger(),
+	}
+
+	releasedIncarnations := []*Incarnation{
+		createTestIncarnation("latest-good", retired, 0),
+		//createTestIncarnation("latest-failing", failing, 100), // failing incarnations do not get returned by releasable()
+		createTestIncarnation("old", retired, 0),
+	}
+	releasedIncarnations[0].fastRelease()
+	releasedIncarnations[0].status.Scale.Current = 1
```

Comment:
why do these get set here in the test? a `retired` revision would never have a nonzero current scale, right?

Reply:
You're right about the scale. I should not be setting this here. `fastRelease()` does happen though, when there are no other good releases.